plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,489 @@
+    type: str
+  query:
+    description:
+      - Query given path for selected query attributes from
+         RouterOS aip and return '.id'.
+          WHERE is key word which extend query. WHERE format is
+         key operator value - with spaces.
+          WHERE valid operators are "==", "!=", ">", "<".
+          Example path "ip address" and query ".id address" will return
+         only ".id" and "address" for all items in "ip address" path.
+          Example path "ip address" and
+          query ".id address WHERE address == 1.1.1.3/32"
+         will return only ".id" and "address" for items in "ip address"
+          path, where address is eq to 1.1.1.3/32.
+          Example path "interface" and query "mtu name WHERE mut > 1400" will
+         return only interfaces "mtu,name" where mtu is bigger than 1400.
+          Equivalent in RouterOS cli "/interface print where mtu > 1400"
+    type: str
+  cmd:
+    description:
+      - Execute any/arbitrary command in selected path,
+         after the command we can add ".id".
+         Example path "system script" and cmd "run .id=*03"
+         is equivalent in RouterOS cli "/system script run number=0",
+          example path "ip address" and cmd "print"
+         equivalent in RouterOS cli "/ip address print"
+    type: str
+'''
+
+EXAMPLES = '''
+---
+- name: routeros_api test
+  hosts: localhost
+  gather_facts: no
+  vars:
+    hostname: "ros_api_hostname/ip"
+    username: "admin"
+    password: "secret_password"
+
+    path: "ip address"
+
+    nic: "ether2"
+    ip1: "1.1.1.1/32"
+    ip2: "2.2.2.2/32"
+    ip3: "3.3.3.3/32"
+
+    addips:
+      - "address={{ ip1 }} interface={{ nic }}"
+      - "address={{ ip2 }} interface={{ nic }}"
+
+    rmips:
+      - "{{ ip2 }}"
+      - "{{ ip3 }}"
+  tasks:
+    - name: get "{{ path }} print"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+      register: print_path
+
+    - name: result "{{ path }} print"
+      debug:
+        msg: '{{ print_path }}'
+
+    - name: add "ip address add {{ addips[0] }}" "ip address add {{ addips[1] }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        add: "{{ item }}"
+      loop: "{{ addips }}"
+      register: addout
+
+    - name: result routeros '.id' for new added items
+      debug:
+        msg: '{{ addout }}'
+
+    - name: query for ".id" in "{{ path }} WHERE address == {{ ip2 }}"
+      routeros_api:
+        hostname: "{{ hostname }}"
+        password: "{{ password }}"
+        username: "{{ username }}"
+        path: "{{ path }}"
+        query: ".id address WHERE address == {{ ip2 }}"
+      register: queryout
+
+    - name: result query result and set fact with '.id' for {{ ip2 }}
+      debug:
+        msg: '{{ queryout }}'
+
+    - set_fact:
```

Comment:
```suggestion
    - ansible.base.set_fact:
```

Reply:
could you please change other stuff similarly to i wrote above?

Reply:
I believe we want "ansible.builtin." not "ansible.base." or I miss something

Reply:
Yes, it should be `ansible.builtin.`.